eval/mt_bench/components.py

```diff
@@ -0,0 +1,178 @@
+# type: ignore
+# pylint: disable=no-value-for-parameter,import-outside-toplevel,import-error
+from typing import List, NamedTuple
+from kfp.dsl import component, Input, Output, Artifact, Model, importer
+from utils.consts import PYTHON_IMAGE
+
+EVAL_IMAGE = "quay.io/sallyom/instructlab-ocp:eval"
+
+@component(base_image=EVAL_IMAGE, packages_to_install=["vllm"])
+def run_mt_bench_op(
+    models_path_prefix: str,
+    models_list: List[str],
+    mt_bench_output: Output[Artifact],
+    merge_system_user_message: bool,
+    # generate_answers,judgment uses a magic word for its mt_bench evaluator  - `auto`
+    # with `auto`, number of gpus allocated for serving is calculated based on environment
+    # https://github.com/instructlab/eval/blob/main/src/instructlab/eval/mt_bench.py#L36
+    max_workers: str = "auto",
+    device: str = None,
+) -> NamedTuple('outputs', best_model=str, best_score=float):
+
+
+    def launch_vllm_server_background(model_path: str, gpu_count: int, retries: int = 60, delay: int = 5):
+        import subprocess
+        import sys
+        import time
+        import requests
+
+        if gpu_count > 0:
+            command = [sys.executable, "-m", "vllm.entrypoints.openai.api_server", "--model", "{model_path}", "--tensor-parallel-size", "{gpu_count}"]
```

Comment:
```suggestion
            command = [sys.executable, "-m", "vllm.entrypoints.openai.api_server", "--model", model_path, "--tensor-parallel-size", gpu_count]
```

Remove the double quote since this is a variable and don't use `{}` since we are not formating strings anymore.

Reply:
Right - I see the error in the run: So this is what I have now:
```
        if gpu_count > 0:
            command = [
                sys.executable,
                "-m", "vllm.entrypoints.openai.api_server",
                "--model", model_path,
                "--tensor-parallel-size",
            ]
        else:
            command = [
                sys.executable,
                "-m", "vllm.entrypoints.openai.api_server",
                "--model", model_path,
            ]

        subprocess.Popen(args=command)
```

Reply:
testing now, hopefully this works!

Reply:
this worked, all set